src/cmd/usb-util.js

```diff
@@ -123,9 +124,16 @@ async function executeWithUsbDevice({ args, func, dfuMode = false } = {}) {
 		res = await func(device);
 	} finally {
 		if (deviceIsProtected) {
-			device = await reopenDevice(device);
-			// FIXME: ignore errors
-			await deviceProtectionHelper.turnOffServiceMode(device);
+			device = await reopenDevice({ id: deviceId });
+			if (!device.isInDfuMode) {
+				device = await waitForDeviceToReboot(device.id);
+			}
+			try {
+				// Only works for 6.1.2 and later
+				await deviceProtectionHelper.turnOffServiceMode(device);
+			} catch (error) {
+				// FIXME: ignore errors
+			}
```

Comment:
I added this to ensure the device is ready for further operations after it finished the respective CLI operation. I forgot which use case needed this change but it was during e2e tests that I found out that this is a good check to have. IMO it adds more robustness. Please let me know if anyone disagrees.
```
if (!device.isInDfuMode) {
        device = await waitForDeviceToReboot(device.id);
}
```